src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts

```diff
@@ -0,0 +1,78 @@
+import { ParsedOrder } from '@cow/modules/limitOrders/containers/OrdersWidget/hooks/useLimitOrdersList'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { BigNumber } from 'bignumber.js'
+import JSBI from 'jsbi'
+
+// TODO: using .toNumber() we potentially lose accuracy
+// TODO: must be refactored with replacing bignumber.js by @ethersproject/bignumber
+function legacyBigNumberToCurrencyAmount(currency: Token, value: BigNumber | undefined): CurrencyAmount<Token> {
+  return CurrencyAmount.fromRawAmount(currency, Math.ceil((value?.toNumber() || 0) * 10 ** currency.decimals))
+}
+
+export function getFilledAmounts(order: ParsedOrder) {
+  const {
+    inputToken,
+    outputToken,
+    kind,
+    feeAmount,
+    executedBuyAmount,
+    executedSellAmount,
+    executedFeeAmount,
+    filledAmount,
+    sellAmount,
+    buyAmount,
+  } = order
+
+  let mainToken: Token
+  let mainAmount: CurrencyAmount<Token>
+  let swappedToken: Token
+  let swappedAmount: JSBI | undefined
+  let action: string
```

Comment:
I know someone that won't be happy with so many `let`

Reply:
Honestly in this case I prefer this instead of having a ternary operator for each variable.